woodwork/datatable.py

```diff
@@ -1023,6 +1023,9 @@ def mutual_information_dict(self, num_bins=10, nrows=None):
         if ks and isinstance(self._dataframe, ks.DataFrame):
             data = data.to_pandas()
 
+        # Dpn't calculate mutual info for columns that aren't unique
+        data = data[[col for col in data.columns if not data[col].is_unique]]
```

Comment:
Can we move the following line out `_handle_nans_for_mutual_info`, and do the following:
```python
        if nrows is not None and nrows < data.shape[0]:
            data = data.sample(nrows)

        # Remove fully null columns
        data = data.loc[:, data.columns[data.notnull().any()]]
        # Remove columns that are unique
        data = data[[col for col in data.columns if not data[col].is_unique]]

        data = self._handle_nans_for_mutual_info(data)
        data = self._make_categorical_for_mutual_info(data, num_bins)
```
- My thought is sample will do the most with large datasets to reduce the size. 
  - After that columns that are all NaN can be removed, and then any that are unique. (The unique column could make for a large computation...)
- Maybe we rename `_handle_nans_for_mutual_info` to `_replace_nans_for_mutual_info`?

Reply:
Yeah, I like this. It also probably makes the behavior of mutual info/what it actually does to the data prior to calculating more clear

Reply:
Question: is it faster to only create the new DataFrames from removing fully null columns and unique columns if there are any columns to remove? 

Right now, if there's nothing to remove we're needlessly copying the DataFrame

Reply:
@tamargrey good point. Yes, we should only do this if there is fully NULL columns and/or unique columns.

Reply:
ok, updated to only do this when necessary